test/integration/extension/lib/shopify/collection-repository/command/ListGraphQlTest.js

```diff
@@ -0,0 +1,23 @@
+const assert = require('assert')
+const StoreFrontApiFactory = require('../../../../../../../extension/lib/shopify/StorefrontClientFactory')
+const ListGraphQl = require('../../../../../../../extension/lib/shopify/collection-repository/command/ListGraphQl')
+const integrationCredentials = require('../../../../../../../.integration-credentials')
+
+describe('ListGraphQl Command', () => {
+  it('should return shopify categories', async () => {
+    const testHandle = 'integration-test-do-no-touch'
+    const storeFrontApiFactory = new StoreFrontApiFactory(integrationCredentials.storefrontAccessToken, integrationCredentials.shopName)
+
+    const shopifyStorefrontClient = storeFrontApiFactory.create()
+
+    const listCommand = new ListGraphQl(shopifyStorefrontClient)
+
+    const categories = await listCommand.execute()
+    const testCategory = categories.find(category => {
+      return category.handle === testHandle
+    })
+
+    assert.equal('Integration-Test *Do no touch*', testCategory.title)
```

Comment:
I would add a test for checking how many entries the list contains

Reply:
Instead of walking through the list to find one specific entry I would check the whole result for equalness. (if you do not need how to achieve this: here is a reference: https://github.com/shopgate/cloud-ext-bigcommerce-catalog/blob/master/test/integration/extension/lib/catalog/category/repository/ShopgateTest.js or here: https://github.com/shopgate/cloud-ext-bigcommerce-catalog/blob/master/test/integration/extension/lib/catalog/product/repository/ShopgateProductListRepositoryTest.js)

Reply:
You should also check id and handle and all other properties. Everything can be wrong and should be tested